Engine/DataFeeds/FileSystemDataFeed.cs

```diff
@@ -502,6 +522,10 @@ private IEnumerator<BaseData> ConfigureEnumerator(SubscriptionRequest request, b
             // optionally apply fill forward logic, but never for tick data
             if (request.Configuration.FillDataForward && request.Configuration.Resolution != Resolution.Tick)
             {
+                var subscriptionConfigs = _subscriptions.Select(x => x.Configuration).Concat(new[] { request.Configuration });
+
+                UpdateFillForwardResolution(subscriptionConfigs);
```

Comment:
This could potentially be done less expensively by checking the current fill forward resolution, and if this subscription is lower and non-tick, replace the value.